### PR TITLE
misc schema changes

### PR DESCRIPTION
**Pull request by gagantrivedi**
resource(feature): Make `type` optional use `STANDARD` as default value
resource(feature): Add default value for  ` default_enabled`
resource(feature): Add `RequiresReplace`  plan modifier to `project_uuid` field

resource(feature_state): Add `RequiresReplace`  plan modifier to `environment_key` field
resource(feature_state): Add `RequiresReplace`  plan modifier to `feature_id` field

resource(multivariate_options): Add `RequiresReplace`  plan modifier to `feature_uuid`

resource(segment): Add `RequiresReplace`  plan modifier to `project_uuid` field

Bump plugin framework to support `default`

### CODE REVIEW

File: flagsmith/models.go [hunk 1]
```diff
@@ -226,6 +224,9 @@ type FeatureResourceData struct {
 
 func (f *FeatureResourceData) ToClientFeature() *flagsmithapi.Feature {
 	typeValue := f.Type.ValueString()
+	if typeValue == "" {
+		typeValue = "STANDARD"
```
Comment (matthewelwell) [inline flagsmith/models.go:228]: I'm not entirely sure why we need this if we're now able to use defaults in terraform?

**Review by matthewelwell — APPROVED** [review]

File: flagsmith/models.go (hunk 1 above)
Comment (khvn26) [reply flagsmith/models.go:228]: Additionally, we can add a `StringInSlice` validator to account for the lack of enum support.

See a good [example](https://github.com/hashicorp/terraform-plugin-sdk/issues/79) of this.

Comment (gagantrivedi) [reply flagsmith/models.go:228]: > I'm not entirely sure why we need this if we're now able to use defaults in terraform?

Yep, this is the fun part, the default value is not available in the state that provider has access to. It's more or less only used in the plan that is shown to the user

Comment (gagantrivedi) [reply flagsmith/models.go:228]: > Additionally, we can add a `StringInSlice` validator to account for the lack of enum support.
> 
> See a good [example](https://github.com/hashicorp/terraform-plugin-sdk/issues/79) of this.

Yeah, good shout

**Pull request merged**